### ISSUE TITLE
Feature request: tfe.Variblae should support Python's float casting

### ISSUE THREAD

**Issue by phizaz** [tensorflow/tensorflow#20816]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.8.0 and 19.0
- **Python version**: 3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
In short, `tfe.Variable` should support:

```
a = tfe.Variable(1.0)
float(a)
```

It is useful in my case, but to state my case it would be very convoluted. If contributors deem appropriate then plaese add a support to this.

**Comment by nairouz** [comment]
It is also useful in my case.

**Comment by robieta** [comment]
I don't think that would be practical. (Since we can't control how a builtin behaved. If you need to change dtypes you will need to use `tf.cast`. Feel free to reopen this issue with more explanation if that doesn't satisfy your use case.

**Comment by shoyer** [comment]
I think this could be supported by implementing the `__float__` method.
